packages/langium/src/grammar/references/grammar-scope.ts

```diff
@@ -141,14 +124,21 @@ export class LangiumGrammarScopeComputation extends DefaultScopeComputation {
         if (container && isAction(node) && node.inferredType) {
             const typeName = getActionType(node);
             if (typeName) {
-                scopes.add(container, {
-                    node,
-                    name: typeName,
-                    type: 'Interface',
-                    documentUri: document.uri,
-                    path: this.astNodeLocator.getAstNodePath(node)
-                });
+                scopes.add(container, this.createInterfaceDescription(node, typeName, document));
             }
         }
     }
+
+    protected createInterfaceDescription(node: AstNode, name: string, document: LangiumDocument = getDocument(node)): AstNodeDescription {
+        const nameNode = this.nameProvider.getNameNode(node) ?? node.$cstNode;
+        return {
+            node,
+            name,
+            segment: nameNode && toDocumentSegment(nameNode),
```

Comment:
This pattern is used quite frequently. Maybe change  `toDocumentSegment` to accept `undefined`? It can be kept backwards-compatible with multiple function signatures (one for defined, one for undefined).